docusaurus/docs/tutorials/build-a-streaming-data-source-plugin.md

```diff
@@ -15,143 +14,321 @@ keywords:
+
+```json
+{
+  ...
+  "id": "grafana-streamingbackendexample-datasource",
+  "metrics": true,
+  "backend": true,
+  ...
+  "streaming": true
+}
+```
+
+## Step 5: Build the plugin
+
+The coding part of the plugin development process is done. Now we need to generate our plugin package to be able to run it in Grafana. Since our plugin has both a frontend and a backend part, we do this in two steps.
+
+1. Build the frontend:
+
+```shell
+npm install
+npm run build
+```
+
+This should download all the dependencies and create the frontend plugin files in the `./dist` directory.
+
+2. Compile the backend code and generate the plugin binaries. For that you should have `mage` installed and then you simply need to run:
+
+```shell
+mage -v
+```
+
+This command compiles the Go code and generate binaries for all Grafana-supported platforms in `./dist`.
+
+## Step 6: Run the plugin
+
+Once the plugin is built, you can copy `./dist` together with its content, rename it to the `id` of your plugin, put it in the plugin's path of your Grafana instance, and test it.
+
+However, there is an easier way of doing all this.
+
+1. Simply run:
+
+```shell
+npm run server
+```
+
+This command runs a Grafana container using docker-compose and puts the built plugins in the right place.
+
+2. To verify the build, go to Grafana at `https://localhost:3000`.
+
+:::note
+
+You may be using Docker Compose instead of docker-compose. If that is your case, you just need to change the command in the `package.json` in the project's root directory.
+
+:::
+
+## Step 7: Test the plugin
+
+1. From Grafana at `https://localhost:3000`, use the default credentials: username `admin` and password `admin`. If you're not presented with a login page, click **Sign in** in the top of the page and insert the credentials.
+
+2. Add the data source. Since we are running in our `docker-compose` environment, we don't need to install it, and it will be directly available for usage. Go to **Connections > Data sources**, using the left menu as shown in the following image:
+
+3. A new page opens, and then click **Add data source**. Grafana opens another page where you can search for the data source name that we've just created.
+
+4. Click the data source's card, and then click **Save & test** on the next page.
+
+5. Optional: Add the following code to `pkg/plugin/datasource.go`. It should replace the `CheckHealth` method that already exists because it was created by the scaffold.
+
+```tsx
+  func (d *Datasource) CheckHealth(_ context.Context, req *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
+    return &backend.CheckHealthResult{
+      Status:  backend.HealthStatusOk,
+      Message: "Data source is working",
+    }, nil
+  }
+```
+
+:::note
+
+If a randomized error occurs, just click **Save & test** until a success message appears:
+
+![Datasource dialog box - Grafana UI.](/img/datasource-dialog.png)
```

Comment:
Since we are adding the `CheckHealth` code in `Step 3: Set up the plugin backend`, we need to remove this.

Reply:
removed